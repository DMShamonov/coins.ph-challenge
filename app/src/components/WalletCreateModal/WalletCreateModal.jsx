// @flow

import * as React from 'react';

import Modal from 'components/Modal';
import WalletForm from 'components/WalletForm';

import type { PropsType } from './types.js.flow';

export default function WalletCreateModal({
  isOpen,
  onClose,
}: PropsType): React.Node {
  return (
    <Modal isOpen={isOpen} contentLabel="Create wallet" onClose={onClose}>
      <WalletForm />
    </Modal>
  );
}
