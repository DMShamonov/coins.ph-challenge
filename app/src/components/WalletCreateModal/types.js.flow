// @flow

export type PropsType = {|
  isOpen: boolean,
  onClose: () => void,
|};
