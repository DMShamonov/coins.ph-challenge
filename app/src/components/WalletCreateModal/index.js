import { connect } from 'react-redux';

import MODAL_ACTION_TYPES from 'redux/actionTypes/modal';
import MODALS from 'constants/modals';

import WalletCreateModal from './WalletCreateModal';

function mapStateToProps({ ui: { modal } }) {
  return {
    isOpen: modal === MODALS.CREATE_WALLET,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onClose() {
      dispatch({ type: MODAL_ACTION_TYPES.ON_HIDE });
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(WalletCreateModal);
