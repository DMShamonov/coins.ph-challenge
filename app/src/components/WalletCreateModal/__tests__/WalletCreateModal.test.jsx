import React from 'react';
import { shallow } from 'enzyme';

import WalletCreateModal from '../WalletCreateModal';

describe('`WalletCreateModal` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<WalletCreateModal isOpen={false} onClose={jest.fn()} />)).toMatchSnapshot();
  });

  test('when `isOpen` prop is `true`', () => {
    expect(shallow(<WalletCreateModal isOpen onClose={jest.fn()} />)).toMatchSnapshot();
  });
});
