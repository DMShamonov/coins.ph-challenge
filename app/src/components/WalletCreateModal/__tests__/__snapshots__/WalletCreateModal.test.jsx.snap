// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`WalletCreateModal\` component should render correctly when \`isOpen\` prop is \`true\` 1`] = `
<Modal
  contentLabel="Create wallet"
  isOpen={true}
  onClose={[MockFunction]}
>
  <ConnectFunction />
</Modal>
`;

exports[`\`WalletCreateModal\` component should render correctly with default props 1`] = `
<Modal
  contentLabel="Create wallet"
  isOpen={false}
  onClose={[MockFunction]}
>
  <ConnectFunction />
</Modal>
`;
